File: python-fundamentals/diagonal-sum.py
def diagonalSum(mat):
    sumList = []
    dim_matrix = len(mat)
    for i in range(0,dim_matrix):
        j = dim_matrix - i - 1
        if i !=j:
            sumList.append(mat[i][i])
            sumList.append(mat[i][j])
        else:
            sumList.append(mat[i][j])
    print(sumList)
    return sum(sumList)


mat = [[7,3,1,9],[3,4,6,9],[6,9,6,6],[9,5,8,5]]

diagonalSum(mat)

File: swift-algos/diagonalSum.swift
class Solution {
    func diagonalSum(_ mat: [[Int]]) -> Int {
        var res = 0
        for i in 0..<mat.count{
            res += mat[i][i]
        }
        for i in 0..<mat.count{
            if mat.count - 1 - i != i {
                res += mat[mat.count - 1 - i][i]
            }
            
        }
        return res
    }
}

let mat = [[1,2,3],
           [4,5,6],    
           [7,8,9]]

let s = Solution()
print(s.diagonalSum(mat))